### ISSUE TITLE
Build error Ubuntu 18.04,  ERROR: Unknown function

### ISSUE THREAD

**Issue by lahtis** [FeralInteractive/gamemode#358]
Gamemode doesn't build when attempting to compile on ubuntu 18.04 

cloned from github then followed build instructions.
expected gamemode to build as it did use to build.

error is: 
Checking for function "pidfd_open": NO
Configuring build-config.h using configuration

lib/meson.build:28:0: ERROR: Unknown function "both_libraries".

**System Info (please complete the following information):**
 - OS and version:  Ubuntu 18.04
 - GameMode Version 1.6.1
 - 
[meson-log.txt](https://github.com/FeralInteractive/gamemode/files/8353820/meson-log.txt)

**Comment by fabiscafe** [comment]
That function is only available with Linux 5.3
https://man.archlinux.org/man/pidfd_open.2#VERSIONS

Ub18.04 does ship with 4.15, so it might be because you have an to old Kernel. Where you need to use the HWE or upgrade to a newer kernel/distro.

**Comment by eli-schwartz** [comment]
> lib/meson.build:28:0: ERROR: Unknown function "both_libraries".

This function was added in Meson 0.46.0, and you are using 0.45.0

This project could make that requirement more specific:
```diff
diff --git a/meson.build b/meson.build
index c4031ad..2939c36 100644
--- a/meson.build
+++ b/meson.build
@@ -4,6 +4,7 @@ project(
     default_options : ['c_std=c11', 'warning_level=3'],
     version: '1.6.1',
     license: 'BSD',
+    meson_version: '>=0.46.0',
 )
 
 am_cflags = [
```

However I don't understand the purpose of #258 @stephanlachnit.

Why not just switch from `shared_library` to `library` and let people choose what they want with the `-Ddefault_library` option?

**Comment by lahtis** [comment]
If you readme the file doesn't say anything about dependencies and their version. How does one even know to look for them? Do I have to dig them out of the source code myself?

**Comment by eli-schwartz** [comment]
LTS typically means you get support from the LTS vendor, but not necessarily from Independent projects which aren't part of Ubuntu.

I don't know what Feral's policy is, but like I said there is an enhancement that can be made to ensure that Meson itself errors out with "hey you need to use a newer version of meson, sorry". So that at least helps communicate the versions of dependencies.

**Comment by lahtis** [comment]
I know what LTS-support means. This was just a criticism of incorrect documentation. Even this bug report would not have been sent if it had been mentioned in the documentation or even in the code.

**Comment by stephanlachnit** [comment]
> However I don't understand the purpose of #258 @stephanlachnit.
>
> Why not just switch from `shared_library` to `library` and let people choose what they want with the `-Ddefault_library` option?

Iirc the reasoning was that Games often link their stuff statically for distribution, and thus you want the loader library static, but the system library only shared. But yes, the loader library could be marked `library` with `default_library=both`.

**Comment by lahtis** [comment]
uname -a
Linux Valvoja-CoolComputer 5.4.0-109-generic #123~18.04.1-Ubuntu SMP Fri Apr 8 09:48:52 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

**Comment by lahtis** [comment]
> That function is only available with Linux 5.3 https://man.archlinux.org/man/pidfd_open.2#VERSIONS
> 
> Ub18.04 does ship with 4.15, so it might be because you have an to old Kernel. Where you need to use the HWE or upgrade to a newer kernel/distro.

 uname -a
Linux Valvoja-CoolComputer 5.4.0-109-generic #123~18.04.1-Ubuntu SMP Fri Apr 8 09:48:52 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
